src/main/java/com/google/cloud/tools/maven/config/AppEngineWebXmlConfigProcessor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.maven.config;
+
+import com.google.cloud.tools.appengine.cloudsdk.Gcloud;
+import com.google.cloud.tools.maven.deploy.AbstractDeployMojo;
+import com.google.common.annotations.VisibleForTesting;
+import java.nio.file.Path;
+
+public class AppEngineWebXmlConfigProcessor implements ConfigProcessor {
+
+  private final Path appengineWebXml;
+  private final Gcloud gcloud;
+  private final ConfigReader configReader;
+
+  /** Config processor for appengine-web.xml based applications. */
+  public AppEngineWebXmlConfigProcessor(
+      Path appengineWebXml, Gcloud gcloud, ConfigReader configReader) {
```

Comment:
The reason for accepting `ConfigReader` and making its methods non-static is for testing, right?